### PR TITLE
add processingStepCalculation type

### PR DESCRIPTION
**Pull request by sorja**
Part of #3

### CODE REVIEW

**Review by SteRiccio — DISMISSED** [review]

File: src/chain/index.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Validation } from 'src/validation/index'
+import { ArenaObject } from 'src/common'
+import { ProcessingStep } from './processingStep'
+
+export interface ChainProps {
+  labels: { [key: string]: string }
```
Comment (minotogna) [inline src/chain/index.ts:6]: don't we have a common labels object? if not, let's add it

File: src/chain/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  descriptions: { [key: string]: string }
+  cycles: Array<string>
+}
+
+export interface Chain extends ArenaObject<ChainProps> {
+  dateCreated: string
+  dateModified: string
+  dateExecuted: string
+  validation: Validation
+  statusExec: string
```
Comment (minotogna) [inline src/chain/index.ts:16]: in arena code we have this:
```
export const statusExec = {
  error: 'error',
  success: 'success',
  running: 'running',
}
```
create an enum

File: src/chain/index.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  scriptCommon: string
+  temporary: boolean
+  processingSteps: Array<ProcessingStep>
```
Comment (minotogna) [inline src/chain/index.ts:19]: alphabetically ordered please

File: src/chain/processingStep.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { ArenaObject } from 'src/common'
+import { ProcessingStepCalculation } from './processingStepCalculation'
+
+export interface ProcessingStepProps {
+  entityUuid?: string
+  categoryUuid?: string
+  virtual: boolean
+  variablesPreviousStep: string
```
Comment (minotogna) [inline src/chain/processingStep.ts:8]: alphabetically ordered

File: src/chain/processingStepCalculation.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { ArenaObject } from 'src/common'
+
+export enum ProcessingStepCalculationPropsType {
+  quantitative = 'quantitative',
+  categorical = 'categorical',
+}
+
```
Comment (minotogna) [inline src/chain/processingStepCalculation.ts:7]: let's call the object:
Chain,
Step,
Calculation

no processing prefix

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/chain/calculation.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { ArenaObject } from 'src/common'
+
+export enum CalculationPropsType {
```
Comment (minotogna) [inline src/chain/calculation.ts:3]: this is CalculationType

File: src/chain/calculation.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  aggregateFn: CalculationPropsAggregateFn
+  formula: string
+}
+
+export interface Calculation extends ArenaObject<CalculationProps> {
+  processingStepUuid: string
+  nodeDefUuid: string
+  index: number
+  script: string
+  temporary: boolean
```
Comment (minotogna) [inline src/chain/calculation.ts:29]: sort alphabetically PLEASEEEEEEEEE

File: src/chain/calculation.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  min = 'min',
+  max = 'max',
+  med = 'med',
+}
+
+export interface CalculationProps {
+  labels: { [key: string]: string }
+  type: CalculationPropsType
+  aggregateFn: CalculationPropsAggregateFn
+  formula: string
```
Comment (minotogna) [inline src/chain/calculation.ts:21]: sort alphabetically PLEASEEEEEEEEE

File: src/chain/calculation.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  categorical = 'categorical',
+}
+
+export enum CalculationPropsAggregateFn {
+  sum = 'sum',
+  avg = 'avg',
+  cnt = 'cnt',
+  min = 'min',
+  max = 'max',
+  med = 'med',
```
Comment (minotogna) [inline src/chain/calculation.ts:14]: sort alphabetically PLEASEEEEEEEEE

File: src/chain/index.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import { Validation } from 'src/validation/index'
+import { ArenaObject } from 'src/common'
+import { ProcessingStep } from './processingStep'
+import { Labels } from 'src/labels/index'
+
+export enum StatusExec {
+  error = 'error',
+  success = 'success',
+  running = 'running',
```
Comment (minotogna) [inline src/chain/index.ts:9]: sort alphabetically PLEASEEEEEEEEE

File: src/chain/index.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+export interface ChainProps {
+  cycles: Array<string>
+  descriptions: { [key: string]: string }
```
Comment (minotogna) [inline src/chain/index.ts:14]: don't we have a common Labels object?

File: src/chain/calculation.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export interface CalculationProps {
+  labels: { [key: string]: string }
```
Comment (minotogna) [inline src/chain/calculation.ts:18]: don't we have a common Labels object ?

File: src/chain/calculation.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import { ArenaObject } from 'src/common'
+
+export enum CalculationPropsType {
+  quantitative = 'quantitative',
+  categorical = 'categorical',
```
Comment (minotogna) [inline src/chain/calculation.ts:5]: sort alphabetically PLEASEEEEEEEEE

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/chain/calculation.ts (hunk 7 above)
Comment (sorja) [reply src/chain/calculation.ts:29]: ok...

File: src/chain/index.ts (hunk 10 above)
Comment (sorja) [reply src/chain/index.ts:9]: ok...

File: src/chain/calculation.ts (hunk 9 above)
Comment (sorja) [reply src/chain/calculation.ts:14]: ok...

File: src/chain/calculation.ts (hunk 8 above)
Comment (sorja) [reply src/chain/calculation.ts:21]: ok...

File: src/category/categoryItem.ts [hunk 14]
```diff
@@ -1,8 +1,9 @@
 import { ArenaObject } from 'src/common'
+import { Labels } from 'src/labels';
```
Comment (minotogna) [inline src/category/categoryItem.ts:2]: don't we have auto prettier on commit?
we do not use ```;```

File: src/chain/index.ts [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+  labels: Labels
+}
+
+export interface Chain extends ArenaObject<ChainProps> {
+  dateCreated: string
+  dateExecuted: string
+  dateModified: string
+  processingSteps: Array<StepProps>
```
Comment (minotogna) [inline src/chain/index.ts:22]: this this an array of Step not StepProps

File: src/category/categoryItem.ts (hunk 14 above)
Comment (minotogna) [reply src/category/categoryItem.ts:2]: please, make sure the commit hook works

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/chain/index.ts (hunk 11 above)
Comment (sorja) [reply src/chain/index.ts:14]: Should `descriptions` be `Labels` too ?
or Descriptions?

File: .husky/pre-commit [hunk 16]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (minotogna) [inline .husky/pre-commit:1]: can't you just run `npx mrm lint-staged` instead of adding all these new files ?
https://prettier.io/docs/en/precommit.html

Comment (sorja) [reply .husky/pre-commit:1]: This didnt work anymore for Husky 5+

**Review by minotogna — APPROVED** [review]

**Pull request merged**